include/gmsh.tcc

```diff
@@ -173,8 +173,11 @@ void GMSH<Tdim, Tvertices>::store_element_vertices() {
   Eigen::VectorXd elementkeyvalues(Tvertices);
   Eigen::VectorXd verticesarray(Tdim * Tvertices);
 
+  const unsigned firstelement = elements_.begin()->first;
+
   //! Iterate through element_
-  for (unsigned i = 0; i < elements_.size(); ++i) {
+  for (unsigned i = 0 + firstelement; i < elements_.size() + firstelement;
```

Comment:
This loop is very confusing. Could you explain the logic behind this loop?

Reply:
in `gmsh`, the element number does not start from 0 or 1. it started at what is defined here as the "firstelement" (in hexCone, it's 2405 out of some 9000 more elements)

Thus, we cannot loop from 0 here, as the first element ID written in the mesh file is 2405. 

I personally like @cw646's looping from first to last element, but if you want to loop through the element size, we have to offset it this way

Reply:
Why not just use a range-based for loop?

Reply:
can we do it? `element_` is a map not a vector... we need to get the values of the `first` right?